docs/server-core/java/migration.mdx

```diff
@@ -0,0 +1,434 @@
+---
+title: Migrating from Legacy Java SDK to Java Core
+sidebar_label: Migration Guide
+keywords:
+  - owner:brock
+last_update:
+  date: 2025-05-12
+---
+
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+# Migrating from Legacy Java SDK to Java Core
+
+This guide will help you migrate from the legacy Java/Kotlin Server SDK to the new Java Core SDK. The Java Core SDK offers significant performance improvements and new features, as it's built on a shared Rust core library.
+
+## Why Migrate?
+
+- **Performance**: Java Core evaluates 3-5x faster than the legacy SDK
```

Comment:
aside: this probably lives in a different document, but we should show the actual benchmark data (and code)